### ISSUE TITLE
add pnpm workspaces info

### ISSUE THREAD

**Issue by futurist** [pnpm/pnpm#3398]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

Want a command like `yarn workspaces info`, but no counterpart command for pnpm.

**Describe the solution you'd like**

Maybe add a command like `pnpm workspaces info`

And if pass `--json`, it can output the json results.

**Comment by gluxon** [comment]
Hey everyone, I got linked here by @zkochan's comment in https://github.com/pnpm/pnpm/issues/4154.

1. I'm wondering if resolving this via the existing `pnpm list` command makes sense? In theory that allows the `--depth`, `--filter`, and `--json` flags to be reused.
2. Alternatively we could close the `pnpm list` feature request I opened in favor of a new `pnpm workspaces info` command like this issue proposes.

I would personally prefer extending `pnpm list` to traverse through workspaces (option 1) since a new `pnpm workspaces info` command would create 2 commands for listing out dependencies, and users might be confused when using the wrong one.

What does everyone think?

**Comment by zkochan** [comment]
I think it is fine to extend the `pnpm list` command. `pnpm workspaces` might be a bad command for pnpm because we don't call the packages in the workspace "workspaces".

**Comment by NullVoxPopuli** [comment]
I just ran in to a need for this today as I was adding pnpm support to my project hopper script: https://github.com/NullVoxPopuli/dotfiles/blob/master/home/shell/bash/gg.sh

**Comment by NullVoxPopuli** [comment]
This is the command I ended up using in bash:
```bash
pnpm ls -r --depth -1 --long --parseable \
  | cut -f1 -d':' \
  | sed "s~$PWD~~g" \
  | sed '/^[[:space:]]*$/d' \
  | sed 's~^/~~g'
```
![image](https://user-images.githubusercontent.com/199018/162646567-c4780143-0adc-4df5-864f-437dbe478a0e.png)

**Comment by karlomedallo** [comment]
+1 for this. My use case is I wanted to get all the workspace package dependencies of an app so that during CI it would know if it needs to run a job or not when changes has been detected from the dependent package

**Comment by gluxon** [comment]
Hey @karlomedallo, would https://github.com/pnpm/pnpm/pull/5863 help for that use case?

**Comment by karlomedallo** [comment]
@gluxon yes, that #5863 will definitely help! I hope that gets merged. Thanks!

**Comment by zkochan** [comment]
This is implemented in current pnpm, so closing.

`pnpm list -r --depth -1 --json` prints every workspace project with its name, version, path and private flag.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
